python/ucxx/ucxx/_lib/libucxx.pyx

```diff
@@ -460,8 +460,7 @@ cdef class UCXAddress():
     def length(self) -> int:
         return int(self._length)
 
-    @property
-    def string(self) -> bytes:
+    def __bytes__(self) -> bytes:
         return bytes(self._string)
```

Comment:
We could add a conversion function for both `bytes` and `str`. That way users can get either

```suggestion
    def __bytes__(self) -> bytes:
        return bytes(self._string)

    def __str__(self) -> str:
        return self._string.decode('utf-8')
```

Reply:
I was going to say that this would need us to duplicate constructors, but [we already do](https://github.com/rapidsai/ucxx/blob/0d29ad849578777d5be9f45c1913ff50e8bbeb04/python/ucxx/ucxx/_lib/libucxx.pyx#L407-L429), so we might as well do this. 😄